### PR TITLE
fix several Java warnings

### PR DESCRIPTION
**Pull request by Bananeweizen**
* silence unchecked casts with generics
* replace Maven FileUtils by Apache IO FileUtils (in test)
* silence "discouraged reference" warning in tests
* silence overridden deprecated method warning
* remove unused variables, imports and exception declarations

### CODE REVIEW

File: target-platform/target-platform.target [hunk 1]
```diff
@@ -97,6 +97,12 @@
 					<version>31.1-jre</version>
 					<type>jar</type>
 				</dependency>
+				<dependency>
+					<groupId>commons-io</groupId>
+					<artifactId>commons-io</artifactId>
+					<version>2.11.0</version>
+					<type>jar</type>
+				</dependency>
```
Comment (HannesWell) [inline target-platform/target-platform.target:105]: Is this necessary for this change?

**Review by HannesWell — COMMENTED** [review]
Looks good in general, but I wonder why the extra dependency is necessary?

File: target-platform/target-platform.target (hunk 1 above)
Comment (Bananeweizen) [reply target-platform/target-platform.target:105]: See second bullet point. Recommended replacement for some of the Maven FileUtils methods is Apache Commons FileUtils (according to their Javadoc).

Comment (HannesWell) [reply target-platform/target-platform.target:105]: Acknowledge.
But it looks like commons-io is already in the TP contributed by some IU-loctions so we don't need that extra dependency.

Comment (HannesWell) [inline target-platform/target-platform.target:105]:
```suggestion
```

**Comment by Bananeweizen** [comment]
BTW: The Eclipse Jenkins build failed because of (to my mind) irrelevant changes for the baseline check. There are 2 points about that:
* Shouldn't we move the -D parameter from the Jenkinsfile to the maven config instead? Otherwise this is absolutely not reproducable locally.
* If I use that parameter locally, there are hundreds of differences. I still feel like the baselining doesn't work correctly depending on whether git automatically converts line endings. If that's the case, the Oomph setup would need to enforce a line ending configuration for the m2e repo which fits to whatever is in the baseline (e.g. if the baseline is/was built by a linux machine, we may need to enforce linux line endings in the git config).

**Comment by HannesWell** [comment]
Yes the baseline-check is not yet perfect. Altough Christoph already submitted some improvements I think there is room for more and I already have some ideas (e.g. derive the text-file extensions from the `.gitattributes`).
That imperfection is the reason why the p2-baseline replacement is only strict in the CI to not let devs on Windows fail their build due to those unrelated line-ending changes.

**Comment by laeubi** [comment]
> The Eclipse Jenkins build failed because of (to my mind) irrelevant changes for the baseline check.

It would help to elaborate why they are "irrelevant", I see a previous baseline failure because of changed content, but thta is nothing different than other builds (just that platform often ignores these warnings (and then gets changes undetected as no one cares) and m2e uses it as an error now.

**Comment by laeubi** [comment]
@Bananeweizen I raised a discussion at P2 about the usefulness of the download-size properties in features:
- https://github.com/eclipse-equinox/p2/discussions/213

**Comment by laeubi** [comment]
Please just open specific issues like:
- https://github.com/eclipse-m2e/m2e-core/issues/1211

so we can investigate how to solve these. If git is concerting line-endings automatically I don't think Eclipse-Settings are a valid source. So if anyone knows how to get the files that get converted using EGit we could probabbly use that information but we should not reimplement it.

**Comment by Bananeweizen** [comment]
> Can you share such an example where it does not work as expected?

Building m2e with ` -Dtycho.p2.baselineMode=failCommon` results in
```
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-p2-plugin:4.0.0-SNAPSHOT:p2-metadata (p2-metadata) on project org.eclipse.m2e.archetype.common: baseline and build artifacts have same version but different contents
[ERROR]    no-classifier: different
[ERROR]       META-INF/MANIFEST.MF: different
[ERROR]          Bundle-Developers: baseline='eu;name="Eugene Kuleshov";organization="Sonatype Inc.";timezone=-5,jvanzyl;name="Jason van Zyl";organization="Sonatype Inc.";timezone=-8,vsiveton;email="vsiveton@apache.org";name="Vincent Siveton";organization=ASF;timezone=-5,ifedorenko;name="Igor Fedorenko";organization="Sonatype Inc.";timezone=-5,akraev;name="Anton Kraev";timezone=-5,hboutemy;name="Hervé Boutemy",krel;name="Luká? K?e?an",bentmann;name="Benjamin Bentmann";organization="Sonatype Inc.",dip;name="Dmitry Platonoff";organization="Sonatype Inc.",rseddon;name="Rich Seddon";organization="Sonatype Inc.",mkleint;name="Milos Kleint";organization="Sonatype Inc.",prapicau;name="Pascal Rapicault";organization="Sonatype Inc.",ptatavu;name="Vlad Tatavu";organization="Sonatype Inc.",rgould;name="Richard Gould";organization="Sonatype Inc.",fbricon;name="Fred Bricon";timezone="+1"' != reactor='eu;name="Eugene Kuleshov";organization="Sonatype Inc.";timezone=-5,jvanzyl;name="Jason van Zyl";organization="Sonatype Inc.";timezone=-8,vsiveton;email="vsiveton@apache.org";name="Vincent Siveton";organization=ASF;timezone=-5,ifedorenko;name="Igor Fedorenko";organization="Sonatype Inc.";timezone=-5,akraev;name="Anton Kraev";timezone=-5,hboutemy;name="HervÃ© Boutemy",krel;name="LukÃ¡Å¡ KÅ?eÄ?an",bentmann;name="Benjamin Bentmann";organization="Sonatype Inc.",dip;name="Dmitry Platonoff";organization="Sonatype Inc.",rseddon;name="Rich Seddon";organization="Sonatype Inc.",mkleint;name="Milos Kleint";organization="Sonatype Inc.",prapicau;name="Pascal Rapicault";organization="Sonatype Inc.",ptatavu;name="Vlad Tatavu";organization="Sonatype Inc.",rgould;name="Richard Gould";organization="Sonatype Inc.",fbricon;name="Fred Bricon";timezone="+1"'
[ERROR]       about.html: different
[ERROR]       about_files/LICENSE-1.1.txt: different
[ERROR]       about_files/LICENSE-2.0.txt: different
[ERROR]       about_files/MPL-1.1.txt: different
[ERROR]       about_files/dom4j-LICENSE.txt: different
[ERROR]       about_files/jdom-LICENSE.txt: different
[ERROR]    classifier-sources: different
[ERROR]       about.html: different
[ERROR]       about_files/LICENSE-1.1.txt: different
[ERROR]       about_files/LICENSE-2.0.txt: different
[ERROR]       about_files/MPL-1.1.txt: different
[ERROR]       about_files/dom4j-LICENSE.txt: different
[ERROR]       about_files/jdom-LICENSE.txt: different
```

**Comment by laeubi** [comment]
Please check if 
- https://github.com/eclipse-tycho/tycho/pull/1997
- https://github.com/eclipse-tycho/tycho/pull/1996

improve the situation for you.

**Comment by HannesWell** [comment]
> Please check if
> 
>     * [Improve finding the right comparator for baseline comparison eclipse-tycho/tycho#1997](https://github.com/eclipse-tycho/tycho/pull/1997)
> 
>     * [Ignore some header in the baseline comparison eclipse-tycho/tycho#1996](https://github.com/eclipse-tycho/tycho/pull/1996)
> 
> 
> improve the situation for you.

I still see similar errors with the baseline plugin for m2e.profiles.core and m2e.apt.core.

File: org.eclipse.m2e.core.tests/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -12,6 +12,6 @@ Require-Bundle: org.eclipse.m2e.tests.common,
  org.eclipse.core.runtime,
  org.eclipse.m2e.maven.runtime
 Import-Package: javax.annotation;version="1.2.0",
- org.apache.commons.io;version="2.11.0"
+ org.apache.commons.io
```
Comment (HannesWell) [inline org.eclipse.m2e.core.tests/META-INF/MANIFEST.MF:15]: @laeubi any idea, why Tycho fails to resolver this package import when it has a package version and the exporting bundle is from a p2-repo (looks like a original bundle), but succeeds with version if the providing bundle is from a Maven-target?

**Review by HannesWell — APPROVED** [review]
This change is good and can be squashed-merged once the build succeeds.
Thank you @Bananeweizen for this contribution. :)

**Pull request merged**